src/hypothesis/strategies.py

```diff
@@ -542,6 +542,41 @@ def frozensets(elements=None, min_size=None, average_size=None, max_size=None):
 
 
 @defines_strategy
+def iterables(elements=None, min_size=None, average_size=None, max_size=None,
+              unique_by=None, unique=False):
+    """This has the same behaviour as lists, but returns iterables instead.
+
+    It is easy to forget that some iterables cannot be indexed (eg sets)
+    and some do not support indexing (eg generators).  This strategy
+    generates iterators, ensuring that you do not accidentally depend on
+    sequence behaviour.
+
+    """
+    class PrettyIter(object):
+        def __init__(self, values):
+            self._values = values
+            self._iter = iter(self._values)
+
+        def __iter__(self):
+            return self._iter
+
+        if PY2:  # pragma: no cover
```

Comment:
I wanted to avoid a visible attribute when running on Py3, but the decorator idea is elegant.